### DISCUSSION TITLE
Setting overleaf credentials

### DISCUSSION THREAD

**Discussion by Atomyka** [rodluger/showyourwork#345] [Q&A]
Hello folks,

thanks for this awesome tool! I'm just trying to get my first Overleaf-integrated repo set up.

I am slightly confused about the part on setting the Overleaf credentials (email + password):
a) Some parts of the documentation refer to both environment _secrets_ (and action _secrets_) and environment _variables_. I might be wrong but I thought variables are publicly visible? So I'd only want to set the credentials as secrets, not variables?
b) I ended up putting the credentials as both environment secrets and action secrets (but not environment variables), as explained in [this Github documentation](https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-an-environment). Then, I executed the setup command with the respective Overleaf ID I wish to use. But then I get the error message:
Fetching Overleaf repo...
Overleaf credentials `OVERLEAF_EMAIL` and/or `OVERLEAF_PASSWORD` not found. These should be set as both environment variables and GitHub repository secrets.
Deleting directory article...

Am I supposed to enter the information somewhere else? I also thought about manually writing the build.yml file that also contains the overleaf credentials (see [bottom of this page](https://show-your.work/en/latest/action/)). But from reading the documentation, I did think the build.yml file is autogenerated by the setup command? 

Any help is appreciated, @rodluger :)

**Comment by eigenpaul** [comment]
Figured it out. Run the following two commands on your machine

`export OVERLEAF_EMAIL="XXX" `
`export OVERLEAF_PASSWORD="YYY" `

**Comment by HealthyPear** [comment]
I will close this discussion as it is obsolete after #467.
